### DISCUSSION TITLE
Vue3 响应式这么强大了。为什么还需要 pinia ？

### DISCUSSION THREAD

**Discussion by pengnima** [vuejs/pinia#1893] [Help and Questions] [answered]
没有贬低的意思，我个人也非常喜欢pinia，小巧精悍，只是确实有如此疑问。如下展示

在Vue3里，可以在 ts 文件里直接使用 reactive ，ref 这些让我们的数据具有响应式。 那么，我是不是可以利用这点去做组件间的相互通信，比如，我在下面声明了一个类，最后返回一个实例，然后在其他兄弟组件间调用，使用。发现响应式的都可以统一响应到。。

当然我知道有官方的状态管理器当然更好，而且 pinia 小巧强大，在工作项目肯定是会使用 pinia 来完成复杂的逻辑交互。但就是刚好在写一些东西的时候，使用了类似下面的这种写法，所以心里一直有个疑问。使用下面这种方式的话 和 pinia 的话，就结果，性能上来说有没有什么区别

![1](https://user-images.githubusercontent.com/44219540/209464572-5dda9261-d726-4243-b96d-fd39181da9f3.png)

![QQ图片20221225183315](https://user-images.githubusercontent.com/44219540/209464596-c5adf0e0-6a68-4679-a9a3-066a4249b45a.png)

![QQ图片20221225183322](https://user-images.githubusercontent.com/44219540/209464650-73af89d2-0fd1-4bb3-b7e5-bc3e1878c54f.png)

![QQ图片20221225183319](https://user-images.githubusercontent.com/44219540/209464660-60c34125-1e15-4098-a83c-3a33cfb2c10d.png)

![image](https://user-images.githubusercontent.com/44219540/209464673-e6c2fd4e-575d-44f5-a7f6-a38d04a8d2d4.png)

**Comment by velly** [comment]
Don't reinvent the wheel. 
It will be a big problem when you're working on large-scale project if you use ref everywhere.

**Reply by pengnima** [reply]
原来如此，谢谢哥们

**Comment by DIYer22** [comment, marked as answer]
好问题，我也和你有一样的疑问。直到找到了这篇文章和讨论：  
(Good question, I have the same question as you. Until I found this article and its discussion)

Summary: 他们俩差不太多，但 pinia支持 SSR(Server-Side Rendering 服务器端渲染)和选项式API

https://juejin.cn/post/7348680291937435682
